### PR TITLE
docs: deprecated.md to sphinx docs

### PR DESCRIPTION
**Pull request by HashedDan**
Signed-off-by: Daniel Mangum <georgedanielmangum@gmail.com>

Description: Moved DEPRECATED.md to sphinx docs.
Risk Level: Low - only documentation
Testing: Compiles with sphinx docs without warnings or errors
Docs Changes: deprecated.rst created in intro section of sphinx docs and added to toctree
Release Notes: N/A
Fixes: #6386

### CODE REVIEW

**Comment by HashedDan** [comment]
@mattklein123 I have updated all links previously referencing DEPRECATED.md. The one place that you may want to take a look is at deprecate_version.py to make sure my updates there are functioning correctly.

**Comment by mattklein123** [comment]
Thanks @HashedDan can you check format CI? I think you might have changed the Python format.

/wait

**Comment by HashedDan** [comment]
Sorry about that @mattklein123! I believe my most recent commit corrected the formatting issue. Thanks!

File: tools/deprecate_version/deprecate_version.py [hunk 1]
```diff
@@ -49,17 +49,17 @@ class DeprecateVersionError(Exception):
 
 # Figure out map from version to set of commits.
 def GetHistory():
-  """Obtain mapping from release version to DEPRECATED.md PRs.
+  """Obtain mapping from release version to docs/root/intro/deprecated.rst PRs.
 
-  Returns:
-    A dictionary mapping from release version to a set of git commit objects.
-  """
+    Returns:
```
Comment (mattklein123) [inline tools/deprecate_version/deprecate_version.py]: nit: can you potentially revert the white space changes in this file?

**Comment by alyssawilk** [comment]
I can do a pass but given I'm not great with python and the bug filing script is the one @htuch wrote he may be a better reviewer.   My best suggestion would be for @HashedDan to fake a token and do a trial run to make sure it works.  If not I'm happy to fix it next week as part of our release process.

**Comment by HashedDan** [comment]
@mattklein123 @alyssawilk Thanks for looking over this! I went back and reverted the whitespace changes then squashed changes into a single commit. Let me know if there is anything else needed!

**Comment by mattklein123** [comment]
@alyssawilk oops sorry I thought you wrote that one. @htuch do you mind taking a quick look?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -51,7 +51,7 @@ maximize the chances of your PR being merged.
   deprecation window. Within this window, a warning of deprecation should be carefully logged (some
   features might need rate limiting for logging this). We make no guarantees about code or deployments
   that rely on undocumented behavior.
-* All deprecations/breaking changes will be clearly listed in [DEPRECATED.md](DEPRECATED.md).
+* All deprecations/breaking changes will be clearly listed in [deprecated log](docs/root/intro/deprecated.rst).
```
Comment (htuch) [inline CONTRIBUTING.md]: Nit: `the [deprecated log]`

Comment (htuch) [reply CONTRIBUTING.md]: Same below in various places.

File: docs/root/intro/deprecated.rst [hunk 3]
```diff
@@ -1,64 +1,68 @@
...
+* Use of `rate_limit_service` configuration in the `bootstrap configuration <https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/bootstrap/v2/bootstrap.proto>`_ is deprecated.
 * Use of `runtime_key` in `RequestMirrorPolicy`, found in
-  [route.proto](https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/route/route.proto)
+  `route.proto <https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/route/route.proto>`_
   is deprecated. Set the `runtime_fraction` field instead.
-* Use of buffer filter `max_request_time` is deprecated in favor of the request timeout found in [HttpConnectionManager](https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto)
-
-## Version 1.8.0 (Oct 4, 2018)
+* Use of buffer filter `max_request_time` is deprecated in favor of the request timeout found in `HttpConnectionManager <https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto>`_
```
Comment (htuch) [inline docs/root/intro/deprecated.rst]: FWIW, the version history is lexicographically sorted, this is (I think) chronological. I think that makes a lot of sense for deprecations, but might be worth calling out as a required style issue in the guidelines.

File: docs/root/intro/deprecated.rst [hunk 4]
```diff
@@ -1,64 +1,68 @@
...
+`Breaking Change Policy <https://github.com/envoyproxy/envoy/blob/master//CONTRIBUTING.md#breaking-change-policy>`_.
 
 The following features have been DEPRECATED and will be removed in the specified release cycle.
 A logged warning is expected for each deprecated item that is in deprecation window.
 
-## Version 1.10.0 (pending)
-* Use of `use_alpha` in [Ext-Authz Authorization Service](https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/auth/v2/external_auth.proto) is deprecated. It should be used for a short time, and only when transitioning from alpha to V2 release version.
+Version 1.10.0 (pending)
+========================
+* Use of `use_alpha` in `Ext-Authz Authorization Service <https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/auth/v2/external_auth.proto>`_ is deprecated. It should be used for a short time, and only when transitioning from alpha to V2 release version.
```
Comment (htuch) [inline docs/root/intro/deprecated.rst]: So, we have some fields here which should now be `:ref:` back to the proto docs. I think it's not a good use of time to fix all of these up, but maybe you could do it for the few in 1.10.0 and add guidance in the docs above about using proper RST links for fields and messages.

**Comment by HashedDan** [comment]
@htuch thanks for your comments! I have made the necessary updates. Please let me know if there is anything else needed!

**Comment by mattklein123** [comment]
@HashedDan big apologies but we cut a release since we started this PR. Can you merge master and reconcile? Thank you!!

/wait

**Comment by HashedDan** [comment]
@mattklein123 no problem at all! Apologies for my delay and congrats on 1.10.0!

**Comment by HashedDan** [comment]
@mattklein123 Okay, I believe everything is now in order. Thanks for your patience!

**Comment by lizan** [comment]
Can we add back DEPRECATED.md to master branch with a link to its new place? Released versions still prints logs like:

```
Using deprecated option 'envoy.api.v2.Cluster.hosts' from file cds.proto. This configuration will be removed from Envoy soon. Please see https://github.com/envoyproxy/envoy/blob/master/DEPRECATED.md for details.
```

And this now results 404.

**Comment by mattklein123** [comment]
> Can we add back DEPRECATED.md to master branch with a link to its new place?

+1 great idea. @HashedDan do you mind doing a follow up PR?

**Comment by HashedDan** [comment]
@mattklein123 Certainly! @lizan thanks for calling this out!

**Pull request merged**